crest/Assets/Crest/Crest/Shaders/OceanInput.hlsl

```diff
@@ -0,0 +1,93 @@
+// Crest Ocean System
+
+// This file is subject to the MIT License as seen in the root of this folder structure (LICENSE)
+
+#ifndef CREST_OCEAN_INPUT_INCLUDED
+#define CREST_OCEAN_INPUT_INCLUDED
+
+#include "OceanConstants.hlsl"
+
+/////////////////////////////
+// Samplers
+
+UNITY_DECLARE_SCREENSPACE_TEXTURE(_CameraDepthTexture);
+UNITY_DECLARE_SCREENSPACE_TEXTURE(_BackgroundTexture);
+
+sampler2D _Normals;
+sampler2D _ReflectionTex;
+samplerCUBE _ReflectionCubemapOverride;
+sampler2D _FoamTexture;
+sampler2D _CausticsTexture;
+
+/////////////////////////////
+// Constant buffer: CrestPerMaterial
+
+CBUFFER_START(CrestInputsPerMaterial)
+half3 _Diffuse;
+half3 _DiffuseGrazing;
+
+half _RefractionStrength;
+half4 _DepthFogDensity;
+
+half3 _SubSurfaceColour;
+half _SubSurfaceBase;
+half _SubSurfaceSun;
+half _SubSurfaceSunFallOff;
+
+half _SubSurfaceDepthMax;
+half _SubSurfaceDepthPower;
+half3 _SubSurfaceShallowCol;
+half3 _SubSurfaceShallowColShadow;
+
+half _CausticsTextureScale;
```

Comment:
should these be behind #ifdef CAUSTICS , and similar questions for other params that are behind shader options? maybe its fine as it is - maybe unity removes unused ones anyway..

Reply:
I think they do. I was just matching what we have downstream. I will add them in here and downstream.